### ISSUE TITLE
Show download speed / amount / time remaining

### ISSUE THREAD

**Issue by iainlane** [canonical/multipass#2196]
**What are you trying to do?**

I'm on a slow mobile internet connection (on a train) at the minute. I'm trying to launch a `focal` VM. But the connection is flaky and I don't know if it's likely to be able to download the image.

**What's your proposed solution?**

Show some combination of the total size / amount remaining to be downloaded / current download rate / estimated time left, so that I know what's going on. Maybe it's flapping up and down (actually, pretty sure it is) - would be good to be able to see that somehow.

Probably wants some design input :-)

**Comment by jimporter** [comment]
Currently, both the GUI and the terminal show a percentage indicating download progress. @amylily1011, do you think we should include more info when downloading images, like ETA?

**Comment by amylily1011** [comment]
@jimporter yes, the current guideline didn't have any specific examples but when we work with a progress bars:
1. Case: Indefinite progress -> we should at least show the current progress and some caveat about the fallback. 
2. Case: Finite progress -> current progress and end (it can be size, step, or other factors), remaining time to finish the download (estimated is fine), download speed (if applicable). 

If we have a fallback triggered such as if it takes longer than 30 mins it renders as fail, we can include it as a small side note as well.

**Comment by jimporter** [comment]
@amylily1011 Thanks for the info. Then I think we'd want to add ETA and download speed in the CLI and GUI when we're fetching images.
